manifests/client/mount.pp

```diff
@@ -1,160 +1,310 @@
+  } else {
+    $_nfsd_port = $nfs::nfsd_port
   }
 
-  if $stunnel !~ Undef {
+  if $nfs_version == 3 {
+    $_stunnel = false
+  } elsif $stunnel !~ Undef {
     $_stunnel = $stunnel
-  }
-  else {
+  } else {
     $_stunnel = $nfs::client::stunnel
   }
 
+  if $stunnel_nfsd_port !~ Undef {
+    $_stunnel_nfsd_port = $stunnel_nfsd_port
+  } else {
+    $_stunnel_nfsd_port = $nfs::stunnel_nfsd_port
+  }
+
+  if $stunnel_socket_options !~ Undef {
+    $_stunnel_socket_options = $stunnel_socket_options
+  } else {
+    $_stunnel_socket_options = $nfs::client::stunnel_socket_options
+  }
+
+  if $stunnel_verify !~ Undef {
+    $_stunnel_verify = $stunnel_verify
+  } else {
+    $_stunnel_verify = $nfs::client::stunnel_verify
+  }
+
+  if $stunnel_wantedby !~ Undef {
+    $_stunnel_wantedby = $stunnel_wantedby
+  } else {
+    $_stunnel_wantedby = $nfs::client::stunnel_wantedby
+  }
+
+  #################################
+  # Configure connection and mount
+  #################################
+
+  if ($nfs_version  == 4) {
+    $_nfs_base_options = "nfsvers=4,port=${_nfsd_port},${options},sec=${sec}"
+  } else {
+    $_nfs_base_options = "nfsvers=3,port=${_nfsd_port},${options}"
+  }
+
+  if $_stunnel {
+    # stunnel only carries TCP
+    $_nfs_options = "${_nfs_base_options},proto=tcp"
+  } else {
+    $_nfs_options = $_nfs_base_options
+  }
+
+  if $_stunnel or ($autodetect_remote and simplib::host_is_me($nfs_server)) {
+    $_remote = "127.0.0.1:${remote_path}"
+  } else {
+    $_remote = "${nfs_server}:${remote_path}"
+  }
+
   nfs::client::mount::connection { $name:
-    nfs_server           => $nfs_server,
-    nfs_version          => $nfs_version,
-    nfs_port             => $port,
-    v4_remote_port       => $v4_remote_port,
-    stunnel              => $_stunnel,
-    stunnel_systemd_deps => $stunnel_systemd_deps,
-    stunnel_wantedby     => $stunnel_wantedby
+    nfs_server             => $nfs_server,
+    nfs_version            => $nfs_version,
+    nfsd_port              => $_nfsd_port,
+    firewall               => $nfs::firewall,
+    stunnel                => $_stunnel,
+    stunnel_nfsd_port      => $_stunnel_nfsd_port,
+    stunnel_socket_options => $_stunnel_socket_options,
+    stunnel_verify         => $_stunnel_verify,
+    stunnel_wantedby       => $_stunnel_wantedby,
+    tcpwrappers            => $nfs::tcpwrappers
   }
 
   if $autofs {
-    include '::autofs'
+    simplib::assert_optional_dependency($module_name, 'simp/autofs')
+    include 'autofs'
 
-    Class['nfs::install'] -> Class['::autofs::install']
+    Class['nfs::install'] -> Class['autofs::install']
 
-    # This is a particular quirk about the autofs service ordering
-    Class['autofs::service'] ~> Service[$::nfs::service_names::rpcbind]
-
-    # Need to handle the wildcard cases
-    $_mount_point = split($name,'wildcard-')[-1]
+    if $autofs_indirect_map_key {
+      $_mount_point = $name
+      $_map_key = $autofs_indirect_map_key
```

Comment:
Based on what I am seeing in simp-autofs module, looks like this should be translated to 'wildcard-$_mount_point' or we will have resource conflicts when multiple nfs::client::mount defines with $autofs_indirect_map_key='*' are defined.

Will fix.